zebra-state/src/request.rs

```diff
@@ -13,14 +13,28 @@ use crate::Response;
 ///
 /// This enum implements `From` for [`block::Hash`] and [`block::Height`],
 /// so it can be created using `hash.into()` or `height.into()`.
-#[derive(Clone, Debug, PartialEq, Eq)]
+#[derive(Copy, Clone, Debug, PartialEq, Eq)]
 pub enum HashOrHeight {
     /// A block identified by hash.
     Hash(block::Hash),
     /// A block identified by height.
     Height(block::Height),
 }
 
+impl HashOrHeight {
+    /// Unwrap the inner height or attempt to retrieve the height for a given
+    /// hash if one exists.
+    pub fn unwrap_height<F>(self, op: F) -> Option<block::Height>
```

Comment:
This name feels a bit unintuitive to me -- I expect "unwrap" to indicate assuming in some way the presence of the value.  Maybe `height_or_else` would fit better with the Rust naming conventions?